cli/cook/querying.py

```diff
@@ -243,19 +243,21 @@ def __get_latest_instance(job):
     raise CookRetriableException(f'Job {job["uuid"]} currently has no instances.')
 
 
-def query_unique_and_run(clusters, entity_ref, command_fn, wait=False):
+def query_unique_and_run(clusters, entity_ref, command_fn, wait=False, do_not_resove_sandbox_directory=False):
     """Calls query_unique and then calls the given command_fn on the resulting job instance"""
 
     def query_unique_and_run():
         query_result = query_unique(clusters, entity_ref)
         if query_result['type'] == Types.JOB:
             job = query_result['data']
             instance = __get_latest_instance(job)
-            directory = mesos.retrieve_instance_sandbox_directory(instance, job)
+            directory = partial(mesos.retrieve_instance_sandbox_directory, instance=instance, job=job)\
+                if do_not_resove_sandbox_directory else mesos.retrieve_instance_sandbox_directory(instance, job)
             command_fn(instance, directory, query_result['cluster'])
         elif query_result['type'] == Types.INSTANCE:
             instance, job = query_result['data']
-            directory = mesos.retrieve_instance_sandbox_directory(instance, job)
+            directory = partial(mesos.retrieve_instance_sandbox_directory, instance=instance, job=job) \
+                if do_not_resove_sandbox_directory else mesos.retrieve_instance_sandbox_directory(instance, job)
```

Comment:
Seems confusing to me that it sometimes returns a function and sometimes returns a value

Reply:
that's the tradeoff for otherwise not refactoring much. do you have a preferred alternative?

Reply:
OK sounds good